server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -205,6 +246,65 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
         }
     }
 
+    private long getPoolInterval(Product prod) {
+        long interval = maxCdkLifeDays;
+        String prodThenString = prod.getAttributeValue("expired_after");
+        if (prodThenString != null && Long.parseLong(prodThenString) < maxCdkLifeDays) {
+            interval = Long.parseLong(prodThenString);
+        }
+        return 1000 * 60 * 60 * 24 * interval;
+    }
+
+    private Pool getCdkPool(Consumer consumer, String sku) {
+        List<Pool> pools = poolCurator.listByConsumer(consumer);
+        for (Pool p : pools) {
+            if (p.hasAttribute(Pool.DEVELOPMENT_POOL_ATTRIBUTE) &&
+                p.hasAttribute(Pool.REQUIRES_CONSUMER_ATTRIBUTE) &&
+                p.getAttributeValue(Pool.REQUIRES_CONSUMER_ATTRIBUTE).equals(consumer.getUuid())) {
+                return p;
+            }
+        }
+        return null;
+    }
+
+    protected Pool assembleDevPool(Consumer consumer, String sku) {
+        // all good. create a pool for the CDK consumer
+        Set<Product> providedProducts = new HashSet<Product>();
+        Date now = new Date();
+
+        Product prod = getCdkInstalledProduct(consumer, sku);
```

Comment:
This is not an installed product, it is the SKU. By reusing this method here, we get an error about missing installed product if it isn't found.